include/llama/View.hpp

```diff
@@ -21,27 +21,19 @@ namespace llama
     {
         template <typename Allocator>
         using AllocatorBlobType = decltype(std::declval<Allocator>()(0));
-
-        template <typename Allocator, typename Mapping, std::size_t... Is>
-        LLAMA_FN_HOST_ACC_INLINE static auto makeBlobArray(
-            const Allocator& alloc,
-            const Mapping& mapping,
-            std::integer_sequence<std::size_t, Is...>) -> Array<AllocatorBlobType<Allocator>, Mapping::blobCount>
-        {
-            return {alloc(mapping.getBlobSize(Is))...};
-        }
     } // namespace internal
 
-    /// Creates a view based on the given mapping, e.g. \ref mapping::AoS or
-    /// \ref mapping::SoA. For allocating the view's underlying memory, the
-    /// specified allocator is used (or the default one, which is \ref
-    /// allocator::Vector). This function is the preferred way to create a \ref
-    /// View.
+    /// Creates a view based on the given mapping, e.g. \ref mapping::AoS or \ref mapping::SoA. For allocating the
+    /// view's underlying memory, the specified allocator callable is used (or the default one, which is \ref
+    /// allocator::Vector). The allocator callable is called with the size of bytes to allocate for each blob of the
+    /// mapping. This function is the preferred way to create a \ref View.
     template <typename Mapping, typename Allocator = allocator::Vector<>>
     LLAMA_FN_HOST_ACC_INLINE auto allocView(Mapping mapping = {}, const Allocator& alloc = {})
         -> View<Mapping, internal::AllocatorBlobType<Allocator>>
     {
-        auto blobs = internal::makeBlobArray<Allocator>(alloc, mapping, std::make_index_sequence<Mapping::blobCount>{});
+        Array<internal::AllocatorBlobType<Allocator>, Mapping::blobCount> blobs;
+        for (int i = 0; i < Mapping::blobCount; i++)
+            blobs[i] = alloc(mapping.getBlobSize(i));
```

Comment:
That might be a pessimization if the allocator's blob type is not cheap to construct.